osgar/lib/depth.py

```diff
@@ -4,13 +4,14 @@
 
 import cv2
 import numpy as np
-import sys
 
 FX = 462.1  # Focal length.
 # Placement of the camera with rspect to the center of the robot.
 CAMX, CAMY, CAMZ = 0.23, 0, (0.19 + 0.06256005)
 # Image dimensions.
 CAMW, CAMH = 640, 360
+# Principal point, position of optical axis
+RX, RY = CAMW / 2. + 0.5, CAMH / 2. + 0.5
```

Comment:
The big question is: Do we want to have it with the same bug as SubT specification of X2, i.e (w+1)/2 and (h+1)/2, or correct, i.e. (w-1)/2 and (h-1)/2?

In any case, given the experiments with drone that has a different RGBD camera, the fully correct solution is to have these parameters in config, different per robot, or to read them from ROS.

Reply:
> The big question is: Do we want to have it with the same bug as SubT specification of X2, i.e (w+1)/2 and (h+1)/2, or correct, i.e. (w-1)/2 and (h-1)/2?

As I understand it, having it in the middle is not correct as long as it is different from the model in SubT. It would be nice to be able to fix it in the model but given the number of PRs that hang in limbo in the SubT repo, I guess there is a zero chance of fixing it there. The only other option is to fix it here.

> In any case, given the experiments with drone that has a different RGBD camera, the fully correct solution is to have these parameters in config, different per robot, or to read them from ROS.

If we ever get there, we can definitely improve along those lines. In the mean time, what I am proposing here is better than what we currently have. Who knows if/when the drone materializes or if we even use this code for it.

Reply:
Ok, let's do it gradually. I agree that the current value is both a) wrong, b) differently wrong than SubT X2. I.e. it is wrong without a good reason. Your change is an improvement.